saleor/graphql/checkout/mutations.py

```diff
@@ -429,19 +433,23 @@ def save(cls, info, instance: models.Checkout, cleaned_input):
         # Set checkout country
         country = cleaned_input["country"]
         instance.set_country(country)
-
         # Create checkout lines
         channel = cleaned_input["channel"]
         variants = cleaned_input.get("variants")
         quantities = cleaned_input.get("quantities")
         if variants and quantities:
-            add_variants_to_checkout(
-                instance,
-                variants,
-                quantities,
-                channel.slug,
-                reservation_length=get_reservation_length(info.context),
-            )
+            try:
+                add_variants_to_checkout(
+                    instance,
+                    variants,
+                    quantities,
+                    channel.slug,
+                    info.context.site.settings.limit_quantity_per_checkout,
+                    reservation_length=get_reservation_length(info.context),
+                )
+            except InsufficientStock as exc:
```

Comment:
Why re-adding this try..except block? `add_variants_to_checkout` should only change the state while validation should happen before in "clean" methods.

Reply:
You're right, I have removed this `try/except` clause.